### DISCUSSION TITLE
How I overcame a few bugs and limitations

### DISCUSSION THREAD

**Discussion by ClandestineClout** [gildas-lormeau/SingleFile#915] [General]
I've overcome several obstacles that until now prevented me from using this in any effective way. 

First, the filename issue (stuff like 56833935-156b-4d8c-a00f-19599c6513d3.html) may be related to an external download manager attempting (but failing) to intercept the download. In my case, disabling the Internet Download Manager Integration Module extension restored the proper filenames saving behavior. 

Actually I only discovered this when trying a fresh profile, when the problem came back immediately after I installed an IDM integration extension to try and solve the next problem: that Edge's native little download list popup kept showing up every time a page was saved. Note that I am NOT referring here to the "Save as" dialogue, just the one that shows you recently downloaded files from Edge's built-in download button in the toolbar and let you click on them.

I couldn't find a way to disable this annoying thing, which I was not ready to live with having auto-save enabled, and did not want to go through the hassle of setting the companion (which I had tried and failed previously) just to save pages without interruption. I thought, there must be another way.

So I looked around to see if there were browser-based download managers that would allow me to prevent that popup from showing up. Indeed, the first extension I tried, ["Download Manager"](https://microsoftedge.microsoft.com/addons/detail/download-manager/phalbpghhjknlmomkmimbamfceiddlic) for Edge and Chrome, did the trick. As soon as I installed it, downloads go directly to it without interruption. 

Now pages are auto-saved after loading with the proper file name and there's no interruption to the viewing area (no recent downloads popup), and the only indicators of the page being saved are one tiny blue badge on the Download Manager extension icon in the toolbar, and the appearance of a green arrow on the native Edge downloads button in the toolbar. These are perfect, barely perceptible peripheral indicators that the extension is working.

The last problem I had was moving the files to a designated folder (for later search/analysis using Recoll, FLP, etc). Since this functionality is only made available with the help of the companion, which I'd really rather not have to set up, I wrote a very quick and simple AutoHotkey script for transferring html files from the downloads folder to a designated location, which I then added to startup (using a separate AHK script I've been using called AddToStartup). Now it checks for html files every 60 seconds and transfers them to a folder called OfflineWeb:

```
Loop{
FileMove, C:\Users\...\Downloads\*html, C:\Users\...\Downloads\OfflineWeb\
Sleep 60000
}
```

I do hope the issue with Internet Download Manager can be solved because I rely on it all the time.

**Comment by gildas-lormeau** [comment]
Thank you very much for your feedback @ClandestineClout!

The filename issue is annoying, I tried to spam Google about it here: https://bugs.chromium.org/p/chromium/issues/detail?id=892133 but it does not help to see it fixed. I guess we have to live with it for the coming years.

I am aware that SingleFile companion is not convenient to install. It's silly because it would have solved your problems. I'm thinking about it regularly and I hope to come up with something more user-friendly.

I'm also moving this issue to the Discussions to keep it visible for everyone.
